src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisScope.cs

```diff
@@ -85,6 +86,22 @@ private AnalysisScope(IEnumerable<SyntaxTree> trees, IEnumerable<AdditionalText>
             Debug.Assert(isPartialAnalysis || FilterSpanOpt == null);
             Debug.Assert(isPartialAnalysis || !isSyntacticSingleFileAnalysis);
 
+            if (filterSpanOpt.HasValue)
+            {
+                Debug.Assert(filterFile.HasValue);
+                Debug.Assert(filterFile.Value.SourceTree != null);
+                Debug.Assert(filterSpanOpt.Value.Length <= filterFile.Value.SourceTree.Length);
```

Comment:
>Debug.Assert(filterSpanOpt.Value.Length <= filterFile.Value.SourceTree.Length); [](http://example.com/codeflow?start=16&length=79)

I am not sue why do we need to rely on this fact and what guarantees the enforcement. If the span comes from an external user, we probably cannot make any assumptions about the value.

Reply:
> I am not sue why do we need to rely on this fact and what guarantees the enforcement. If the span comes from an external user, we probably cannot make any assumptions about the value

Thanks, you are correct, I'll remove this assert.